### PR TITLE
Fix for load screen bug

### PR DESCRIPTION
**Pull request by zakmoCA**
i remembered you talking about this bug ages ago and me saying id check it out 🤙

i thought this would help kickstart things getting back into hopefully regular programming by giving it a go, it felt more motivating then working on my stuff currently. 

wrote up an explanation of the bug in `bug.md` and just general (if you can call it that) guidance for this bug case, might be helpful in early days! think it might have helped me with a project like this earlier on, just based on what ive learnt so far, which isnt a crazy amount. 

excuse the grammar and tone, tried to keep it as dry and straightforward as possible. if you have any questions about it just let me know, happy to go over it! 🙂

### CODE REVIEW

File: running.lua [hunk 1]
```diff
@@ -233,21 +248,21 @@ function running.update(dt) -- dt = delta time
...
         if lure.x <= 0 or lure.x >= WINDOW_WIDTH then -- This means the GAP per creature is 140pixels
             lure.x_direction = lure.x_direction * -1
-        end 
+        end
     end
 
     if love.keyboard.wasReleased("space") and backToFishing then
         -- PLAYER is pulling ('pulled')     
         player.anim = player.animations.pulled
         lure.anim = lure.animations.pulled
```
Comment (ShannyxMP) [inline running.lua:257]: Thanks for taking the time to review the code! After testing the code again, the SPACEBAR bug in the menu state is definitely fixed. However, I am finding that the splashing animation of the lure is being activated continously after the SPACEBAR is released. The other issue is that it's also not initiating the pulling animation when the player releases the SPACEBAR.

Comment (zakmoCA) [reply running.lua:257]: ayooo i think i wasnt familiar enough with the mechanics of the whole game to notice those things were amiss lmao, all looked the same to me i saw menu fixed and said adios lmao, but i see what you mean now.

wasnt sure how you want to approach this new fix, like would you want to give it a go yourself, or you prefer i clean up my mess lmao etc., but ive taken a quick look and wrote some ideas as jumping off point *just* in case you wanted to give this one a crack yourself 🕵️‍♀️  if you prefer im happy to clean up these new bugs and just update doc with report, can do that too just lmk! 

ill double check your desired game behaviour in OG version and play again later, then its just gonna be about moving checks and state vars to their appropriate places 🙂

#### after quick look rn: 
involves just moving those state variables around and resetting them upon state transition to reflect `function love.keyreleased(key)` effect having a new home inside `running.load()` now, if i QA'd by actually playing it after menu fix it would be apparent 😆 but briefly:
1. the problem seems related to how the `lureBeingPulled` state and the `love.keyboard.isDown("space")` and `love.keyreleased("space")` checks are interacting. cleaning up those checks and tracking `lureBeingPulled` state to see if the lure is moving (while space is held) and appropriatedly updating its state should work
2. fix could involve setting `lureBeingPulled = true` when the lure direction (code where the coordinates direction being changed) is updated—i didn’t update that in my PR when i added `downspace` / `backToFishing`  / `lureBeingPulled` conditional after the `--LURE moving back-and-forth along x-axis` logic etc. maybe something like:

```lua
 -- Handle SPACEBAR press/release logic
    if love.keyboard.isDown("space") and `backToFishing` and not lureBeingPulled then
        -- PLAYER is 'snagged'
        player.anim = player.animations.snagged
        lure.anim = lure.animations.snagged

        -- LURE moving back-and-forth along x-axis
        lure.x = lure.x + (lure.x_speed * lure.x_direction)
        if lure.x <= 0 or lure.x >= WINDOW_WIDTH then
            lure.x_direction = lure.x_direction * -1
        end

       -- ❗️ UPDATED CODE BELOW ⬇️
        -- Set pulling state to TRUE
        lureBeingPulled = true
```

- could also add a conditional after that to bring back the pulling animation when spacebar released, maybe something like this:

```lua
 -- Adds to caughtCreatures table if not caught yet
            if not alreadyCaught then
                table.insert(caughtCreatures, newCreature)
            end

            -- Start the display timer when a creature is caught
            creatureDisplayTimer = creatureDisplayDuration
            isTimerRunning = true
        else
            lure.y = lure.y + (lure.y_speed * lure.y_direction)
        end

       -- ❗️ UPDATED CODE BELOW ⬇️
        -- Reset splashing animation after release
        if not love.keyboard.isDown("space") then
            lure.anim = lure.animations.pulled
            lureBeingPulled = false
        end
```


4. also might need to make sure the `love.keyreleased` has the `backToFishing` check so the pulling logic only runs when in the fishing state ie: 

```lua 
function love.keyreleased(key)
    -- .....
    -- .....
    -- .....

    if backToFishing then 
        lureBeingPulled = true
        -- .....
        -- .....
        -- .....

        -- (.....)
    end

    -- .....
    -- .....
end
``` 

missed QA'ing the fix, so i didn’t catch the breaking change 😅 [how i checked the game after the patch lmaooo](https://www.google.com/url?sa=i&url=https%3A%2F%2Fgiphy.com%2Fexplore%2Flazy-security-guard&psig=AOvVaw1EIXTNzT7QXnTH2gJ3CnTX&ust=1735537281618000&source=images&cd=vfe&opi=89978449&ved=0CBMQjRxqFwoTCIDJoauizIoDFQAAAAAdAAAAABAI), anyway...

#### tl;dr solution thoughts rn: 
- deconflict `love.keyboard.isDown("space")` and `love.keyreleased("space")`
- track `lureBeingPulled` when space is held
- and fix the condition cleanup to stop the splash animation from looping endlessly.

dont feel stressed about giving it a go because ive changed your original `running.lua` somewhat and its been ages since you wrote this code and its unfamiliar which might make it too much for late december maybe 😂  im happy to pick back up later and update doc too 🙂

Comment (zakmoCA) [reply running.lua:257]: closing this, easier to start new PR
